### DISCUSSION TITLE
how to access store from register-service-worker.js

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#14927] [CLI - PWA mode] [answered]
I have tried several ways of accessing my vuex store from the `register-service-worker.js` file, but none of them work. Can someone point me to an example for importing a standard quasar vuex store created from `src/store` or `src/store/storename`?

**Comment by ssuess** [comment]
Tried to follow the code here and do similarly, but this seems not to work with all the changes in Quasar v2: https://github.com/quasarframework/quasar/discussions/10288

**Reply by metalsadman** [reply]
that's pretty common use case and proven to work `outside ssr`, so better show what you've done yourself.

**Reply by ssuess** [reply]
Sure, here is my previous code:
```
import { createStore } from 'vuex'
import VuexPersistence from 'vuex-persist'
import localForage from 'localforage'
import VuexReset from '@ianwalter/vuex-reset'

import rstruth from './rstruth'

const vuexLocal = new VuexPersistence({
  key: 'APP_Key',
  storage: localForage,
  asyncStorage: true,
  supportCircular: true,
  reducer: (state) => JSON.parse(JSON.stringify(state))
})

export default function (/* { store } */) {
  const Store = createStore({
    modules: {
      rstruth
    },
    plugins: [vuexLocal.plugin, VuexReset()]
  })
  return Store
}
```
and following the recommendations in the linked post, I changed it to:
```
import { createStore } from 'vuex'
import VuexPersistence from 'vuex-persist'
import localForage from 'localforage'
import VuexReset from '@ianwalter/vuex-reset'

import rstruth from './rstruth'

const vuexLocal = new VuexPersistence({
  key: 'APP_Key',
  storage: localForage,
  asyncStorage: true,
  supportCircular: true,
  reducer: (state) => JSON.parse(JSON.stringify(state))
})
let store
export default function (/* { store } */) {
  const Store = createStore({
    modules: {
      rstruth
    },
    plugins: [vuexLocal.plugin, VuexReset()]
  })
store = Store  
return Store
}
export { store } // also tried just export store
```
then when trying to use `import { store } from '../src/store'`  I get `undefined`

**Reply by ssuess** [reply]
and if I try `import store from '../src/store'` I get `ƒ tu(){const e=(t={modules:{rstruth:Zc},plugins:[Yc.plugin,Xc()]},new dc(t));var t;return eu=e,e}`

**Reply by metalsadman** [reply]
Try some console logs in your store and register-service-worker file, see which gets called first. Try commenting your vuex plugins too.

**Reply by ssuess** [reply]
- Console logs in both files, vuex log shows up before register-service-worker log
- Removed vuex plugins. Same result.

Any other suggestions?

**Reply by ssuess** [reply]
Interestingly, if I try to console log `store` inside my store index, it is ALSO `undefined`:
```
import { createStore } from 'vuex'
import VuexPersistence from 'vuex-persist'
import localForage from 'localforage'
import VuexReset from '@ianwalter/vuex-reset'

import rstruth from './rstruth'

console.log('this is vuex file')

const vuexLocal = new VuexPersistence({
  key: 'APP_Key',
  storage: localForage,
  asyncStorage: true,
  supportCircular: true,
  reducer: (state) => JSON.parse(JSON.stringify(state))
})
let store
export default function (/* { store } */) {
  const Store = createStore({
    modules: {
      rstruth
    },
    plugins: [vuexLocal.plugin, VuexReset()]
  })
store = Store  
return Store
}
console.log('this is vuex store', store)
export { store } // also tried just export store
```

**Reply by ssuess** [reply]
UGH, I think I finally see why. If I do this in my `register-service-worker.js` file, I get the data:

`setTimeout(() => { console.log('store', store.state.rstruth) }, 1500)`

race conditions. The bane of my existence.

**Comment by ssuess** [comment, marked as answer]
As mentioned in comment above, answer is: race condition

> UGH, I think I finally see why. If I do this in my `register-service-worker.js` file, I get the data:
> 
> `setTimeout(() => { console.log('store', store.state.rstruth) }, 1500)`
> 
> race conditions. The bane of my existence.
